src/Perl6/Actions.nqp

```diff
@@ -1538,19 +1538,32 @@ class Perl6::Actions is HLL::Actions does STDActions {
 
     # Produces a LoL from a semicolon list
     method semilist($/) {
-        if $<statement> {
+        if $<statement> -> $statements {
             my $past := QAST::Stmts.new( :node($/) );
-            if nqp::elems($<statement>) > 1 {
+            if nqp::elems($statements) > 1 {
                 my $l := QAST::Op.new( :name('&infix:<,>'), :op('call') );
-                for $<statement> {
+                for $statements {
                     my $sast := $_.ast || QAST::WVal.new( :value($*W.find_single_symbol('Nil')) );
                     $l.push(wanted($sast, 'semilist'));
                 }
                 $past.push($l);
                 $past.annotate('multislice', 1);
             }
             else {
-                $past.push($<statement>[0].ast || QAST::WVal.new( :value($*W.find_single_symbol('Nil')) ));
+                my $ast := $statements[0].ast;
+
+                # anything like "||foo" ?
+                if nqp::istype($ast,QAST::Op) && $ast.name eq '&prefix:<|>' {
+                    if $*W.lang-ver-before("e") {
```

Comment:
Any particular reason why this check is before the next if? Once .e is released (and going forward), it'd be checking the version any time it stumbles upon a `|`, regardless of whether it is `||` or not. I guess checking for the version is very cheap, but still, wouldn't it be better to check the version inside the next `if`?

Reply:
Addressed with 8dcba57368